brion/io/morphologies.cpp

```diff
@@ -0,0 +1,136 @@
+/* Copyright (c) 2013-2015, EPFL/Blue Brain Project
+ *                          Juan Hernando <jhernando@fi.upm.es>
+ *
+ * This file is part of Brion <https://github.com/BlueBrain/Brion>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "morphologies.h"
+#include "morphology.h"
+
+#include "brion/circuit.h"
+#include "brion/blueConfig.h"
+
+#include <lunchbox/log.h>
+
+namespace brion
+{
+namespace io
+{
+namespace morphologies
+{
+
+typedef std::map<std::string, boost::shared_ptr< Morphology > > MorphologyMap;
+
+Morphologies load( const BlueConfig& config, const GIDSet& gids )
+{
+    const brion::Circuit circuit( config.getCircuitSource( ));
+    return load( circuit, config.getMorphologySource(), gids );
+}
+
+Morphologies load( const BlueConfig& config, GIDSet& gids )
+{
+    brion::Circuit circuit( config.getCircuitSource( ));
+    return load( circuit, config.getMorphologySource(), gids );
+}
```

Comment:
I simply raised it since I did the same for the Publisher( uri ) ctor, which @tribal-tec reverted for good reasons.
There are alternatives, mostly by using different function signatures (return a std::pair, provide an error function, ...)